src/legacy/integrations/contact-input-integration-wrapper.tsx

```diff
@@ -0,0 +1,52 @@
+/*
+ * SPDX-FileCopyrightText: 2023 Zextras <https://www.zextras.com>
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+import React, { useMemo } from 'react';
+
+import {
+	type ContactChipAction,
+	ContactInput,
+	ContactInputItem,
+	type ContactInputProps
+} from './contact-input';
+import { useActionEditDL } from '../../actions/edit-dl';
+import { DistributionList } from '../../model/distribution-list';
+
+export type ContactInputIntegrationWrapperProps = Omit<ContactInputProps, 'contactActions'> & {
+	actions?: ContactInputProps['contactActions'];
+};
+
+const isDistributionList = (
+	contact: DistributionList | ContactInputItem
+): contact is DistributionList => 'isOwner' in contact;
+
+export const ContactInputIntegrationWrapper = ({
+	actions,
+	...rest
+}: ContactInputIntegrationWrapperProps): React.JSX.Element => {
+	const actionEditDL = useActionEditDL();
+
+	const defaultActions = useMemo(
```

Comment:
If an user of this integration want to add other actions to the default one, is it possible?

Reply:
#### Short answer
Actions are set the same as before. External modules which use the integration must declare the additional actions in the chips (`value` field).

#### Long story
We had to extract this action because we had a dependency cycle, and we had to hide the edit dl action when the contact input is used inside the edit dl modal. With the changes we made, we implicitly fixed a problem of the integration about the custom actions: all custom actions were previously lost when the rendered chip was a DL one. To give you an example, the duplicated icon wasn't visible when the duplicated contact was a dl.

So, yes, it is still possible to add custom actions on the chips of the contact input when using the integration, and we added some tests for this wrapper component to check it.